Updates to dev docs + Recommonmark -> MyST Parser (#1082)
jupyterhub/repo2docker: docs/source/contributing/tasks.md

and maintaining repo2docker. If you'd like more guidance for how
to do these things, reach out in the [JupyterHub Gitter channel](https://gitter.im/jupyterhub/jupyterhub).

:::{admonition} Set up your development environment first!
:class: tip
Before attempting most tasks, follow the directions to [set up your development environment](contributing:local-dev).

> Though note that many of these tasks to **not** require Docker.

Reviewer: ```suggestion
Though note that many of these tasks do **not** require Docker.
```

   python freeze.py
   ```

   This script will resolve dependencies and write them to the respective `.lock`
   files.

3. After the freeze script finishes, a number of files will have been **modified**.
   They roughly follow this pattern:

   ```
   repo2docker/buildpacks/conda/environment*
   ```

   You should **commit all modified files** to git.

4. Make a Pull Request to the `jupyterhub/repo2docker` repository, with a description
   of what versions were bumped / what new packages were added and why. If you fix a
   bug or add new functionality consider adding a new test to prevent the bug from
   coming back/the feature breaking in the future.

> 5. Once the pull request is approved (but not yet merged), Update the
>    change log (details below) and commit the change log, then update
>    the pull request.

Author: ```suggestion
```

OK this removes it!

   ```

>    This script will resolve dependencies and write them to the respective `.frozen.yml`

Reviewer: ```suggestion
   This script will resolve dependencies and write them to the respective `.lock`
```

   files.

3. After the freeze script finishes, a number of files will have been **modified**.
   They roughly follow this pattern:

   ```
   repo2docker/buildpacks/conda/environment*
   ```

   You should **commit all modified files** to git.

4. Make a Pull Request to the `jupyterhub/repo2docker` repository, with a description
   of what versions were bumped / what new packages were added and why. If you fix a
   bug or add new functionality consider adding a new test to prevent the bug from
   coming back/the feature breaking in the future.

5. Once the pull request is approved (but not yet merged), Update the
   change log (details below) and commit the change log, then update

>    the pull request.

Reviewer: I've never done this changelog step, and I've never seen anyone else do it either. Should we remove it?

Author: +1 from me if it's not being followed

Reviewer 2: I don't think anyone does this and we eventually gave up on having a changelog that is accurate, so let's remove it :(
